public/app/features/auth-config/fields.ts

```diff
@@ -66,52 +141,194 @@ export const fieldMap: Record<string, FieldData> = {
   apiUrl: {
     label: 'API Url',
     type: 'text',
+    description:
+      'The user information endpoint of your OAuth2 provider. Information returned \n' +
+      'by this endpoint must be compatible with OpenID UserInfo.',
     validation: {
       required: false,
     },
   },
   roleAttributePath: {
     label: 'Role Attribute Path',
+    description: 'JMESPath expression to use for Grafana role lookup.',
+    type: 'text',
+    validation: {
+      required: false,
+    },
+  },
+  name: {
+    label: 'Display name',
+    description: 'Helpful if you use more than one identity providers or SSO protocols',
+    type: 'text',
+  },
+  allowSignUp: {
+    label: 'Allow sign up',
+    description: 'If not enabled, only existing Grafana users can log in using OAuth',
+    type: 'switch',
+  },
+  autoLogin: {
+    label: 'Auto login',
+    description: 'Log in automatically, skipping the login screen',
+    type: 'switch',
+  },
+  signoutRedirectUrl: {
+    label: 'Sign out redirect URL',
     type: 'text',
     validation: {
       required: false,
     },
   },
+  emailAttributeName: {
+    label: 'Email attribute name',
+    description: 'Name of the key to use for user email lookup within the attributes map of OAuth2 ID token.',
+    type: 'text',
+  },
+  emailAttributePath: {
+    label: 'Email attribute path',
+    description: 'JMESPath expression to use for user email lookup from the user information.',
+    type: 'text',
+  },
+  nameAttributePath: {
+    label: 'Name attribute path',
+    description:
+      'JMESPath expression to use for user name lookup from the user ID token. \n' +
+      'This name will be used as the user’s display name.',
+    type: 'text',
+  },
+  loginAttributePath: {
+    label: 'Login attribute path',
+    description: 'JMESPath expression to use for user login lookup from the user ID token.',
+    type: 'text',
+  },
+  idTokenAttributeName: {
+    label: 'ID token attribute name',
+    description: 'The name of the key used to extract the ID token from the returned OAuth2 token.',
+    type: 'text',
+  },
+  roleAttributeStrict: {
+    label: 'Role attribute strict mode',
+    description: 'If enabled, denies user login if the Grafana role cannot be extracted using Role attribute path',
+    type: 'switch',
+  },
+  allowAssignGrafanaAdmin: {
+    label: 'Allow assign Grafana admin',
+    description: 'If enabled, it will automatically sync the Grafana server administrator role',
+    type: 'switch',
+  },
+  skipOrgRoleSync: {
+    label: 'Skip organization role sync',
+    description: 'Prevent synchronizing users’ organization roles from your IdP',
+    type: 'switch',
+  },
+  defineAllowedGroups: {
+    label: 'Define Allowed Groups',
+    type: 'switch',
+  },
+  defineAllowedTeamsIds: {
+    label: 'Define Allowed Teams Ids',
+    type: 'switch',
+  },
+  usePkce: {
+    label: 'Use Pkce',
+    description: 'If enabled, it will automatically sync the Grafana server administrator role',
```

Comment:
Can we add links to the description of the label?
```suggestion
    description: 'If enabled, Grafana will use [Proof Key for Code Exchange (PKCE)](https://datatracker.ietf.org/doc/html/rfc7636) with the OAuth2 Authorization Code Grant.',
```